src/main/java/com/epam/aidial/core/controller/DownloadFileController.java

```diff
@@ -29,37 +27,29 @@ protected Future<?> handle(ResourceDescription resource, boolean hasWriteAccess)
             return context.respond(HttpStatus.BAD_REQUEST, "Can't download a folder");
         }
 
-        Future<Blob> blobFuture = proxy.getVertx().executeBlocking(() ->
-                proxy.getStorage().load(resource.getAbsoluteFilePath()), false);
+        Future<FileService.FileStream> blobFuture = proxy.getVertx().executeBlocking(() ->
+                fileService.getFile(resource), false);
 
         Promise<Void> result = Promise.promise();
-        blobFuture.onSuccess(blob -> {
-            if (blob == null) {
+        blobFuture.onSuccess(fileStream -> {
+            if (fileStream == null) {
                 context.respond(HttpStatus.NOT_FOUND);
                 result.complete();
                 return;
             }
 
-            Payload payload = blob.getPayload();
-            MutableContentMetadata metadata = payload.getContentMetadata();
-            String contentType = metadata.getContentType();
-            Long length = metadata.getContentLength();
-
             HttpServerResponse response = context.getResponse()
-                    .putHeader(HttpHeaders.CONTENT_TYPE, contentType)
+                    .putHeader(HttpHeaders.CONTENT_TYPE, fileStream.contentType())
                     // content-length removed by vertx
-                    .putHeader(HttpHeaders.CONTENT_LENGTH, length.toString());
-
-            try {
-                InputStreamReader stream = new InputStreamReader(proxy.getVertx(), payload.openStream());
-                stream.pipeTo(response, result);
-                result.future().onFailure(error -> {
-                    stream.close();
-                    context.getResponse().reset();
-                });
-            } catch (IOException e) {
```

Comment:
why you removed this catch block? are you sure that in this case future will be handled correctly?

Reply:
No longer thrown by the code.